psydac/linalg/stencil.py

```diff
@@ -16,14 +16,31 @@
 __all__ = ['StencilVectorSpace','StencilVector','StencilMatrix', 'StencilInterfaceMatrix']
 
 #===============================================================================
-def compute_diag_len(p, md, mc, return_padding=False):
-    n = ((np.ceil((p+1)/mc)-1)*md).astype('int')
-    ep = np.minimum(0, n-p)
-    n = n-ep + p+1
+def compute_diag_len(pads, shifts_domain, shifts_codomain, return_padding=False):
+    """ Compute the diagonal lenght of the stencil matrix for each direction,
+        using the shifts of the domain and the codomain.
+        Parameters
+        ----------
+        pads : tuple-like (int)
+         Padding along each direction
+
+        shifts_domain : tuple_like (int)
+         Shifts of the domain along each direction
+
+        mshifts_codomainc : tuple_like (int)
+         Shifts of the codomain along each direction
+
+        return_padding : bool
+            Return the new padding if True
+        """
```

Comment:
Nice docstring!

Please add the `Returns` section, which is missing, and fix a few typos:
- lenght -> length (swap positions of `t` and `h`)
- mshifts_codomainc -> shifts_codomain (remove leading `m` and trailing `c`)

Also, please add an empty line before the `Parameters` section.

Reply:
done